tests/numpy/test_generic.py

```diff
@@ -171,3 +175,94 @@ def test_query_dataframe(self):
         self.assertEqual(
             'Extras for NumPy must be installed', str(e.exception)
         )
+
+
+class DataFrameDBAPITestCase(NumpyBaseTestCase):
+    types = \
+        'a int64, b string, c datetime,' \
+        'd fixed_string(10), e decimal(9, 5), f float64,' \
+        'g low_cardinality(string), h nullable(int32)'
+
+    columns = 'a,b,c,d,e,f,g,h'
+    data = [
+        [
+            123, 'abc', datetime(2024, 5, 20, 12, 11, 10),
+            'abcefgcxxx', Decimal('300.42'), 3.402823e12,
+            '127001', 332
+        ],
+        [
+            456, 'cde', datetime(2024, 6, 21, 12, 13, 50),
+            '1234567890', Decimal('171.31'), -3.4028235e13,
+            '127001', None
+        ],
+        [
+            789, 'efg', datetime(1998, 7, 22, 12, 30, 10),
+            'stream sql', Decimal('894.22'), float('inf'),
+            '127001', None
+        ],
+    ]
+
+    @contextmanager
+    def query_data(self):
+        with self.create_stream(self.types):
+            with connect('proton://localhost') as conn:
+                with conn.cursor() as cur:
+                    cur.execute(
+                        f'INSERT INTO test ({self.columns}) VALUES',
+                        self.data
+                    )
+                    cur.execute(f'SELECT {self.columns} FROM test')
+
+                    yield cur
+
+    def test_dbapi_fetchnumpy(self):
+        try:
```

Comment:
tests are usually executed in controlled environment where libs such as numpy are installed. Can you make sure CI not skip these tests? 

Besides, we can safely remove the try / except as the tests are not released and user will have no impact without numpy installation.